src/components/Editor.jsx

```diff
@@ -137,6 +143,14 @@ class Editor extends React.Component {
     this._editor.moveCursorTo(0, 0);
     this._resizeEditor();
   }
+
+  _handleKeyPress(event) {
+    if (
+      event.key === 'i' && event.metaKey &&
+      !event.altKey && !event.ctrlKey) {
+      this.props.onAutoFormat();
```

Comment:
Hey so I’m not sure if this is an issue outside of Linux, but on my laptop when I do meta-I it indents the text but also inserts an `i` character at the end of the code. Relatedly, I think the command should probably be ctrl-I on Windows and Linux, and Command-I on Mac OS (which I am guessing it already is).

Playing around with it, I think we can call `preventDefault()` on the event, which deals with the insertion issue. However I wonder if it would make more sense to use ACE’s [`commands.addCommand()`](https://ace.c9.io/demo/keyboard_shortcuts.html) method, which adds the keyboard shortcut at the ACE level. It looks like that would make it easier to semantically specify different Mac and non-Mac keyboard shortcuts, and I’m guessing would implicitly handle the character-insertion bug. WDYT?

(Weirdly that method is not actually mentioned in the ACE documentation as far as I can tell, but it definitely exists and the linked example is on the ACE site, so it seems legit to use)

Reply:
ace way seems good i'll look into it

Reply:
got it working, mind testing on linux?